pkg/commands/edit/set/setimage.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"errors"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/pkg/commands/kustfile"
+	"sigs.k8s.io/kustomize/pkg/fs"
+	"sigs.k8s.io/kustomize/pkg/image"
+)
+
+type setImageOptions struct {
+	imageMap map[string]image.Image
+}
+
+// errors
+
+var (
+	errImageNoArgs      = errors.New("no image specified")
+	errImageInvalidArgs = errors.New(`invalid format of image, use one of the following options:
+- <image>=<newimage>:<newtag>
+- <image>=<newimage>@<newtag>
+- <image>:<newtag>
+- <image>@<digest>`)
+)
+
+const separator = "="
+
+// newCmdSetImage sets the new names, tags or digests for images in the kustomization.
+func newCmdSetImage(fsys fs.FileSystem) *cobra.Command {
+	var o setImageOptions
+
+	cmd := &cobra.Command{
+		Use:   "image",
+		Short: `Sets images and their new names, new tags or digests in the kustomization file`,
+		Example: `
+The command
```

Comment:
Can we separate this example? One is for `<image>=<newimage>:<newtag>` and `<image>=<newimage>@<newtag>`? The other is for `<image>:<newtag>` and `<image>@<digest>`.

Reply:
sounds good,  I also added  the case for `<image>=<newimage>` which is also supprted